cms/djangoapps/contentstore/views/library.py

```diff
@@ -168,18 +172,56 @@ def library_blocks_view(library, response_format):
         prev_version = library.runtime.course_entry.structure['previous_version']
         return JsonResponse({
             "display_name": library.display_name,
-            "library_id": unicode(library.course_id),
+            "library_id": unicode(library.location.library_key),
             "version": unicode(library.runtime.course_entry.course_key.version),
             "previous_version": unicode(prev_version) if prev_version else None,
             "blocks": [unicode(x) for x in children],
         })
 
+    can_edit = has_studio_write_access(user, library.location.library_key)
+
     xblock_info = create_xblock_info(library, include_ancestor_info=False, graders=[])
-    component_templates = get_component_templates(library, library=True)
+    component_templates = get_component_templates(library, library=True) if can_edit else []
 
     return render_to_response('library.html', {
+        'can_edit': can_edit,
         'context_library': library,
         'component_templates': json.dumps(component_templates),
         'xblock_info': xblock_info,
-        'templates': CONTAINER_TEMPATES
+        'templates': CONTAINER_TEMPATES,
+        'lib_users_url': reverse_library_url('manage_library_users', unicode(library.location.library_key)),
+    })
+
+
+def manage_library_users(request, library_key_string):
```

Comment:
I think this is still missing unit test coverage. Take a look at the diff cover report and make sure all the appropriate things are being covered for this PR.

https://jenkins.testeng.edx.org/job/edx-platform-report/8147/Diff_Coverage_Report/